packages/web/src/eval.js

```diff
@@ -0,0 +1,32 @@
+export class Dioxus {
+  constructor(sendCallback) {
+    this.sendCallback = sendCallback;
+    this.received = [];
+  }
+  // Receive message from Rust
+  recv() {
+    return new Promise((resolve, _reject) => {
+      // Ever 50 ms check for new data
+      let timeout = setTimeout(() => {
```

Comment:
Instead of polling for new data could we first check if there is any currently resolved data. If there is we can resolve the promise immediately, otherwise we could set a pendingPromise field on the object to the resolve callback. That callback could then be called in the rustSend function.

Reply:
Much better solution! I knew a timeout didn't feel right.